### PR TITLE
Refactors summaries, adds new ActionView Component for Dashboard Item

### PR DESCRIPTION
**Pull request by cdmo**
#60 

Also had to bump ActionView Components in order to get access to new `render?` method.

I did also try out rendering an additional ActionView Component from within an ActionView Component for the little label and number pairs, but it didn't seem worth it.

### CODE REVIEW

**Comment by cdmo** [comment]
Meant to add a preview 

Screenshot 2020-01-24 15 57 03

**Comment by banukutlu** [comment]
> Meant to add a preview
> 
> Screenshot 2020-01-24 15 57 03

this looks so nice!

File: app/views/summaries/_summary.html.erb [hunk 1]
```diff
@@ -6,36 +6,29 @@
 <% end %>
 
 <div class="page-section">
-  <h2>Summary</h2>
+  <h1>Summary</h1>
+  <div class="row">
+    <%= render DashboardItemComponent, model: 'Fines',
+                                       count: patron.fines.sum(&:owed_amount),
```
Comment (banukutlu) [inline app/views/summaries/_summary.html.erb]:
```suggestion
                                       count: number_to_currency patron.fines.sum(&:owed_amount),
```

Comment (banukutlu) [reply app/views/summaries/_summary.html.erb]: wondering if you removed `number_to_currency` on purpose?

File: app/components/dashboard_item_component.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    content_tag :ul, facts, nil, false
+  end
+
+  def item_link
+    url = url_for(controller: @model.to_s.downcase, action: 'index')
+
+    link_to "See all #{@model}", url, class: 'btn btn-primary float-right'
+  end
+
+  def render?
```
Comment (banukutlu) [inline app/components/dashboard_item_component.rb]: 😎  very nice to be able to move this logic to the class

File: spec/components/dashboard_item_component_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                                    count: 2
+                                  },
+                                  {
+                                    label: 'overdue',
+                                    count: 3
+                                  }
+                                ] }}
+
+  it 'makes a list' do
+    component = render_inline(described_class, dashboard_item_args).to_html
```
Comment (banukutlu) [inline spec/components/dashboard_item_component_spec.rb]: could we add component with a `let` like below and remove that each line from the examples?

`let(:component) { render_inline(described_class, dashboard_item_args).to_html }`

I tried it, seemed to be working but I might be missing sth as well? What do you think?

Comment (cdmo) [reply spec/components/dashboard_item_component_spec.rb]: yes, great idea. Also, I refactored reduce Mystery Guest

**Comment by cdmo** [comment]
Now I remember why I didn't add `number_to_currency`! `count` needs to be an integer (or "positve-able". If one could easily coerce a currency into an integer (i.e., with `to_i`) then I wouldn't have needed to add a new argument: `currency`.

**Pull request merged**